docker/dev/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM python:3.6-alpine
+
+RUN apk update && apk add python3-dev build-base git
+
+WORKDIR /app
+
+# Cache the dependencies
+COPY setup.py .
+RUN mkdir xain_fl && \
+    printf '__version__ = "0.2.0"\n__short_version__ = "0.2"' > xain_fl/__version__.py && \
+    touch README.md && \
+    python setup.py egg_info && \
+    cat *.egg-info/requires.txt | grep -v '^\[' | sort | uniq | pip install -r /dev/stdin
+
```

Comment:
Feel free to copy setup.py, cat/grep the version of gRPC, pip install it, and then install the full set of dependencies. They will cache the built package of grpc. Otherwise this is too confusing, specially for an open source project, with a lot of magic constants that make this Dockefile harder to read than necessary

Reply:
That's what I did initially but for some reason gRPC was getting re-installed when I installed `xain-fl` anyway, and I was not able to understand why. When installing _all_ the dependencies at once, things are not getting re-installed. I can add a more informative comment about what this does though.

Reply:
I added a comment to explain what's going on and why we're doing it that way.